src/main/kotlin/uk/gov/dluhc/printapi/rds/mapper/CertificateMapper.kt

```diff
@@ -0,0 +1,38 @@
+package uk.gov.dluhc.printapi.rds.mapper
+
+import org.mapstruct.Mapper
+import org.mapstruct.Mapping
+import org.springframework.beans.factory.annotation.Autowired
+import uk.gov.dluhc.printapi.dto.EroManagementApiEroDto
+import uk.gov.dluhc.printapi.mapper.SourceTypeMapper
+import uk.gov.dluhc.printapi.messaging.models.SendApplicationToPrintMessage
+import uk.gov.dluhc.printapi.rds.entity.Certificate
+import uk.gov.dluhc.printapi.rds.entity.PrintRequest
+import uk.gov.dluhc.printapi.service.IdFactory
+
+@Mapper(uses = [SourceTypeMapper::class, PrintRequestMapper::class, InstantMapper::class])
+abstract class CertificateMapper {
+
+    @Autowired
+    protected lateinit var idFactory: IdFactory
+
+    @Autowired
+    protected lateinit var printRequestMapper: PrintRequestMapper
+
+    @Mapping(target = "id", ignore = true)
+    @Mapping(target = "vacNumber", expression = "java( idFactory.vacNumber() )")
+    @Mapping(source = "localAuthority", target = "issuingAuthority")
+    @Mapping(target = "printRequests", expression = "java( toPrintRequestList(message, ero) )")
+    abstract fun toCertificate(
+        message: SendApplicationToPrintMessage,
+        ero: EroManagementApiEroDto,
+        localAuthority: String
+    ): Certificate
+
+    protected fun toPrintRequestList(
```

Comment:
Couldn't get MapStruct to do this automagically..

Reply:
Try this AfterMapping 

```
@Autowired
    protected lateinit var idFactory: IdFactory

    @Autowired
    protected lateinit var printRequestMapper: PrintRequestMapper

    @Mapping(target = "id", ignore = true)
    @Mapping(target = "vacNumber", expression = "java( idFactory.vacNumber() )")
    @Mapping(source = "localAuthority", target = "issuingAuthority")
    abstract fun toCertificate(
        message: SendApplicationToPrintMessage,
        ero: EroManagementApiEroDto,
        localAuthority: String
    ): Certificate

    @AfterMapping
    protected fun addPrintRequestToCertificate(
        message: SendApplicationToPrintMessage,
        ero: EroManagementApiEroDto,
        @MappingTarget certificate: Certificate
    ) {
        certificate.addPrintRequest(printRequestMapper.toPrintRequest(message, ero))
    }
```

Reply:
Thanks @vishalgupta-valtech that works, and yes it should be using `add` 👍

Reply:
The only difference with using aftermapping is that you can actually call `.addPrintRequest` of Certificate. However IMHO even your impl is ok but just creates a new list of print requests (doesn't add to it)